tool build failure: YAML::RegEx::m_z may be used uninitialized in this function

```
$ make
[  1%] Building CXX object third_party/yaml-cpp/CMakeFiles/yaml-cpp.dir/src/regex_yaml.cpp.o
In file included from /home/mcmaster/prjxray/third_party/yaml-cpp/src/regex_yaml.cpp:1:0:
/home/mcmaster/prjxray/third_party/yaml-cpp/src/regex_yaml.h: In constructor ‘YAML::RegEx::RegEx(const string&, YAML::REGEX_OP)’:
/home/mcmaster/prjxray/third_party/yaml-cpp/src/regex_yaml.h:31:20: error: ‘<anonymous>.YAML::RegEx::m_z’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
 class YAML_CPP_API RegEx {
                    ^~~~~
cc1plus: all warnings being treated as errors
third_party/yaml-cpp/CMakeFiles/yaml-cpp.dir/build.make:518: recipe for target 'third_party/yaml-cpp/CMakeFiles/yaml-cpp.dir/src/regex_yaml.cpp.o' failed
make[2]: *** [third_party/yaml-cpp/CMakeFiles/yaml-cpp.dir/src/regex_yaml.cpp.o] Error 1
CMakeFiles/Makefile2:1873: recipe for target 'third_party/yaml-cpp/CMakeFiles/yaml-cpp.dir/all' failed
make[1]: *** [third_party/yaml-cpp/CMakeFiles/yaml-cpp.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

```
$ gcc --version
gcc (Debian 7.3.0-5) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ cmake --version
cmake version 3.9.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,9 +9,6 @@ if(NOT CMAKE_BUILD_TYPE)
 		FORCE)
 endif()
 
-set(CMAKE_CXX_STANDARD 14)
-add_compile_options(-Wall -Werror)
-
 # Hack for missing option in cctz
 option(BUILD_TESTING "" OFF)
 
@@ -29,6 +26,10 @@ add_subdirectory(third_party/yaml-cpp EXCLUDE_FROM_ALL)
 target_include_directories(yaml-cpp PUBLIC
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/third_party/yaml-cpp/include>
 	)
+
+# Set the CXX standard and compile time for our code only.
+set(CMAKE_CXX_STANDARD 14)
+add_compile_options(-Wall -Werror)
 
 add_subdirectory(lib)
 add_subdirectory(tools)

